benchmark/framework/executor.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Executor to run and measure resources consumed by python code."""
+import datetime
+import inspect
+import time
+from collections import namedtuple
+from multiprocessing import Process, Queue
+from operator import attrgetter
+from statistics import mean
+from typing import Any, Callable, List, Tuple
+
+import memory_profiler  # type: ignore
+
+import psutil  # type: ignore
+
+from tests.common.utils import timeit_context
+
+
+ResourceStats = namedtuple("ResourceStats", "time,cpu,mem")
+
+
+class ExecReport:
+    """Process execution report."""
+
+    def __init__(
+        self,
+        args: tuple,
+        time_passed: float,
+        stats: List[ResourceStats],
+        killed: bool,
+        period: float,
+    ):
+        """Make an instance."""
+        self.args = args
+        self.report_created = datetime.datetime.now()
+        self.time_passed = time_passed
+        self.stats = stats
+        self.killed = killed
+        self.period = period
+
+    def __getattr__(self, attr: str) -> Any:
```

Comment:
Is overriding the default __getattr__ dunder method a common pattern?

Reply:
replaced with more obvious properties: cpu, mem